ibis/backends/tests/test_client.py

```diff
@@ -1629,3 +1624,60 @@ def test_from_connection(con, top_level):
     new_con = backend.from_connection(getattr(con, CON_ATTR.get(con.name, "con")))
     result = int(new_con.execute(ibis.literal(1, type="int")))
     assert result == 1
+
+
+@pytest.mark.notimpl(
+    ["flink"], raises=com.IbisError, reason="not yet implemented for Flink"
+)
+def test_no_accidental_cross_database_table_load(con_create_database):
+    con = con_create_database
+
+    # Create an extra database
+    con.create_database(dbname := gen_name("dummy_db"))
+
+    # Create table with same name in current db and dummy db
+    con.create_table(
+        table := gen_name("table"), schema=(sch1 := ibis.schema({"a": "int"}))
+    )
+
+    con.create_table(table, schema=ibis.schema({"b": "string"}), database=dbname)
+
+    # Can grab table object from current db:
+    t = con.table(table)
+    assert t.schema().equals(sch1)
+
+    con.drop_table(table)
+
+    allowed_exceptions = (
+        com.IbisError,
+        # these exception types are None when the backend dependency that
+        # defines them is not installed
+        *filter(
+            None,
+            (
+                ClickHouseDatabaseError,
+                PySparkAnalysisException,
+                MySQLProgrammingError,
+                ExaQueryError,
+                SnowflakeProgrammingError,
+                GoogleNotFound,
+            ),
+        ),
+        # datafusion really needs to get their exception story in order
+        #
+        # we only want to allow base Exception when we're testing datafusion
+        # otherwise any exceptions, including those that are unrelated to the
+        # problem under test will be considered correctly raising
+        *((Exception,) * (con.name == "datafusion")),
+    )
```

Comment:
gnarly

Reply:
I can make this a bit less nested if that would help

Reply:
Unraveled it a bit